### PR TITLE
Autreg 130 wait for metadata table to be created

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: dynamo/src/main/java/no/bibsys/db/RegistryManager.java [hunk 1]
```diff
@@ -75,50 +78,62 @@ private Registry createRegistryTable(String validationSchemaTable, Registry regi
...
-        if (!tableDriver.tableExists(schemaTable)) {
-            logger.info(
-                    "Schema table does not exist, creating new one, schemaTable={}", schemaTable);
-            tableDriver.createRegistryMetadataTable(schemaTable);
+    private void checkIfRegistryMetadataTableExistsOrCreate(String metadataTable) throws RegistryMetadataTableBeingCreatedException {
+        if (!tableDriver.tableExists(metadataTable)) {
+            logger.info("Registry metadata table does not exist, creating new one, metadataTable={}", metadataTable);
+            tableDriver.createRegistryMetadataTable(metadataTable);
+            try {
+                validateRegistryExists(metadataTable);
```
Comment (brinxmat) [inline dynamo/src/main/java/no/bibsys/db/RegistryManager.java]: Twice?

File: dynamo/src/main/java/no/bibsys/db/RegistryManager.java [hunk 2]
```diff
@@ -75,50 +78,62 @@ private Registry createRegistryTable(String validationSchemaTable, Registry regi
...
+                validateRegistryExists(metadataTable);
+            }catch(RegistryUnavailableException | RegistryNotFoundException e ) {
```
Comment (brinxmat) [inline dynamo/src/main/java/no/bibsys/db/RegistryManager.java]: Linting, generally :)

File: dynamo/src/main/java/no/bibsys/db/exceptions/RegistryMetadataTableBeingCreatedException.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package no.bibsys.db.exceptions;
+
+public class RegistryMetadataTableBeingCreatedException extends Exception {
+
+    private static final long serialVersionUID = 1796783855770289416L;
```
Comment (brinxmat) [inline dynamo/src/main/java/no/bibsys/db/exceptions/RegistryMetadataTableBeingCreatedException.java]: Do we really want to do this?

File: cypress/support/commands.js [hunk 4]
```diff
@@ -10,37 +10,37 @@
 //
 //
 // -- This is a parent command --
-// Cypress.Commands.add("login", (email, password) => { ... })
+// Cypress.Commands.add('login', (email, password) => { ... })
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: fixed

File: cypress/support/commands.js [hunk 5]
```diff
@@ -63,34 +63,44 @@ function waitUntilRegistryIsReady(registryName, count){
...
+			failOnStatusCode: false,
 			headers: {
 				'api-key': apiAdminApiKey,
-				"content-type": "application/json"
+				'content-type': 'application/json'
 			}
 		}).then((response) => {
-			cy.log('api-key: ' + response.body.apiKey)
-			cy.wrap(response.body.apiKey).as('registryAdminApiKey')
+			if(response.status === 503){
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: fixed

File: cypress/support/commands.js [hunk 6]
```diff
@@ -63,34 +63,44 @@ function waitUntilRegistryIsReady(registryName, count){
...
+				const newCount = count + 1;
+				if(newCount < 5){
+					cy.wait(2000)
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: fixed

File: cypress/support/commands.js [hunk 7]
```diff
@@ -63,34 +63,44 @@ function waitUntilRegistryIsReady(registryName, count){
...
+					createRegistry(registryName, apiAdminApiKey, metadataFile, createEntity, newCount)
+				}
+			}else{
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: fixed

File: cypress/support/commands.js [hunk 8]
```diff
@@ -63,34 +63,44 @@ function waitUntilRegistryIsReady(registryName, count){
...
 
-			cy.registryReady(registryName)
-			
-			if(createEntity){
-				cy.log('creating test entity')
-				cy.get('@registryAdminApiKey').then((registryAdminApiKey) => {
-					let testDataFile = 'entityTestData.json'
-					cy.createEntity(registryName, registryAdminApiKey, testDataFile)
-				})
+				if(createEntity){
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: No

File: dynamo/src/main/java/no/bibsys/db/RegistryManager.java (hunk 1 above)
Comment (Eiriknil) [reply dynamo/src/main/java/no/bibsys/db/RegistryManager.java]: fixed

File: cypress/support/index.js [hunk 9]
```diff
@@ -57,15 +57,15 @@ afterEach(function(){
...
+	const statusUrl = '/registry/' + registryName + '/status'
 	cy.log('waiting...')
 	cy.log('counter: ' + count)
 
 	cy.request({
 		url: statusUrl,
 		failOnStatusCode: false
 	}).then(function (response){
-		if(response.status === 303){
+		if(response.status === 303||response.status === 503){
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/index.js]: fixed

File: cypress/support/commands.js [hunk 10]
```diff
@@ -63,34 +63,44 @@ function waitUntilRegistryIsReady(registryName, count){
...
+					cy.log('creating test entity')
+					cy.get('@registryAdminApiKey').then((registryAdminApiKey) => {
+						let testDataFile = 'entityTestData.json'
```
[the remark this answers is not in the document]
Comment (Eiriknil) [reply cypress/support/commands.js]: changed to const

**Pull request merged**